src/mminou.h

```diff
@@ -93,28 +98,258 @@ extern flag g_commandFileSilentFlag; /* For SUBMIT ... /SILENT */
 extern FILE *g_input_fp;  /*!< File pointers */
 extern vstring g_input_fn, g_output_fn;  /*!< File names */
 
-/*! print2 returns 0 if the user has quit the printout.
-  \warning:  never call print2 with string longer than PRINTBUFFERSIZE - 1  */
+/*!
+ * \fn flag print2(const char* fmt,...)
+ * \brief formatted output of a single line with optional page-wise display.
+ *
+ * It is important to understand that the submitted parameters result in a
+ * NUL-terminated string that contains at most one LF character, only allowed
+ * in final position.  A long line exceeding \ref g_screenWidth is broken down
+ * into multiple lines using a built-in line wrap algorithm.  But this must
+ * never be preempted by preparing parameters accordingly.
+ *
+ * Although the output of a single line is the main goal of this function, it
+ * does a lot on the side, each effect individually enabled or disabled by
+ * various global variables that are honored, sometimes even updated.  We skim
+ * through these effects here in short:
+ * 
+ * 1. __Data embedding__.  The \p fmt parameter can contain simple text, that
+ *     is displayed as is.  Or embedded placeholders are replaced with data
+ *     pointed to by the following parameters.  If necessary, data are
+ *     converted to strings before insertion.
+ * 2. Supporting __page-wise display__.  Output on the virtual text display is
+ *     stopped after showing \ref g_screenHeight lines, called a __page__ here.
+ *     The user can read the text at rest, and then request resuming output
+ *     with a short command.  In fact, this feature is enhanced by maintaining
+ *     a \ref pgBackBuffer, allowing the user to recall previously shown pages,
+ *     and navigate them freely forward and backwards.  This feature
+ *     temporarily grabs user input and interprets short commands as navigating
+ *     instructions.  For more see \ref pgBackBuffer.
+ * 3. __Line wrapping__.  When the line to display exceeds the limits in
+ *     \ref g_screenWidth, line breaking is applied.  The wrapping is actually
+ *     done in \ref printLongLine which in turn uses separate print2 calls to
+ *     output each individual line.  This requires a minimum synchronization of
+ *     both the 'master' and 'slave' calls, so that relevant state is carried
```

Comment:
The master and slave terminology here is confusing. Could you define your terms or use different words like "caller" and "callee"?